src/server/error.h

```diff
@@ -35,11 +35,12 @@ using facade::kWrongTypeErr;
 
 #ifndef RETURN_ON_BAD_STATUS
 
-#define RETURN_ON_BAD_STATUS(x) \
-  do {                          \
-    if (!(x)) {                 \
-      return (x).status();      \
-    }                           \
+#define RETURN_ON_BAD_STATUS(x)  \
+  do {                           \
+    OpStatus __s = (x).status(); \
+    if (__s != OpStatus::OK) {   \
+      return (x).status();       \
+    }                            \
```

Comment:
Hi! It looks like you didin't fix that and still have double evaluation in case of failed operation.
I think it should be
```cpp
    if (__s != OpStatus::OK) {   \
      return __s; \
    }  
```

Reply:
🤦🏼 🤦🏼 🤦🏼  Good catch @aleexf !